splinepy/spline.py

```diff
@@ -1319,9 +1319,12 @@ def todict(self, tolist=False):
             should_copy = True
             # attr are either list or np.ndarray
             # prepare list if needed.
-            if isinstance(tmp_prop, np.ndarray) and tolist:
-                tmp_prop = tmp_prop.tolist()  # copies
-                should_copy = False
+            if tolist:
+                if isinstance(tmp_prop, np.ndarray):
+                    tmp_prop = tmp_prop.tolist()  # copies
+                    should_copy = False
+                if p == "knot_vectors":
+                    tmp_prop = [t.tolist() for t in tmp_prop]
```

Comment:
What structure is knot vectors before?

Reply:
it was a list of tracked arrays, but tracked arrays are not serialisable which let to problems with the json module

Reply:
Ok but should we not add the if statement that if the type is tracked array we do this kind of conversion `tolist` to also have any other variables save that might have a tracked array? Maybe also in the future?

Reply:
Can we add a tolist() function to the tracked array?

Reply:
this is a very specific application, so I don't think that this will be necessary. All other variables (ctps, degrees,..) are either list already or are stored as tracked-arrays, which are also interpret-able as numpy ndarrays

Reply:
yes, as it inherits from numpy

Reply:
oh ok now I understand tracked arrays already have that functionality but knot_vector is List[TrackedArray].

Yeah ok that is fine than.